### DISCUSSION TITLE
UnityTechnologies.Unity installs as separate versions - Winget installs updates multiple times

### DISCUSSION THREAD

**Discussion by Master-Guy** [microsoft/winget-pkgs#16038] [Ideas]
When installing UnityTechnologies.Unity packages, it does not uninstall the previous version. This is part of the way it should be working in this specific case. However, winget doesn't deal with this properly: For each version of UnityTechnologies.Unity that is already installed (and not up-to-date), winget installs the newest version once. This means that if you already have 3 out-of-date versions, it will download and install the most recent version 3 times.

Would it be possible to use only the latest installed version of each distinct installed ID?

**Comment by denelon** [comment]
I believe there is a chance with the [Upgrade](https://github.com/microsoft/winget-cli/blob/b94e128bdf773e185b87fce985d213c9d82ed742/schemas/JSON/manifests/v1.0.0/manifest.installer.1.0.0.json#L121) or the [UpgradeBehavior](https://github.com/microsoft/winget-cli/blob/b94e128bdf773e185b87fce985d213c9d82ed742/schemas/JSON/manifests/v1.0.0/manifest.installer.1.0.0.json#L147) node being properly configured some of these cases can be addressed with today's client and schema.

**Reply by Master-Guy** [reply]
If I read it correctly, you are suggesting to automatically uninstall the old version through winget? Having multiple versions of Unity is a core part of the application, so I don't think it would be a good solution.

**Reply by denelon** [reply]
@Master-Guy thanks for the additional context. I think I have a better understanding of the scenario now. This is an interesting situation. I'm not sure yet how we might address with `winget upgrade --all` or with the "pin" feature just yet. This might require some additional design work. 

Let me try to make sure I have the scenario correct.

I have a package <foo>. This package can have multiple versions installed at the same time. When I run `winget upgrade --all` it attempts to install the latest version for each version of <foo> on my machine. I want to be able to somehow indicate that the earlier versions of <foo> are all "OK" and should remain, but I would like the latest version of <foo> added to my machine.

Is that correct?

**Reply by Master-Guy** [reply]
> I have a package . This package can have multiple versions installed at the same time. When I run `winget upgrade --all` it attempts to install the latest version for each version of on my machine. I want to be able to somehow indicate that the earlier versions of are all "OK" and should remain, but I would like the latest version of added to my machine.
> 
> Is that correct?

Correct.
Also, you don't want to download or run the same installer multiple times in a single flow of `winget upgrade`

**Reply by denelon** [reply]
https://github.com/microsoft/winget-cli/issues/1105
